### ISSUE TITLE
Install Production on Ubuntu/Docker

### ISSUE THREAD

**Issue by kang28ivan** [chaskiq/chaskiq#204]
Hello,

A few days ago I successful to install on heroku, but I remove app after a few days of testing because I want to install the application on my own server.
But the problem here is that I've tried a number of installation methods https://dev.chaskiq.io/en/collections/getting-started, none of which have really worked and the error is the same like this

```
root@platform:~/chaskiq# docker-compose run rails rake db:create
Starting chaskiq_postgres_1 ... done
Starting chaskiq_redis_1    ... done
FATAL:  role "postgres" does not exist
Couldn't create 'postgres' database. Please check your configuration.
rake aborted!
ActiveRecord::NoDatabaseError: FATAL:  role "postgres" does not exist
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb:50:in `rescue in postgresql_connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb:33:in `postgresql_connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:887:in `new_connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:931:in `checkout_new_connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:910:in `try_to_checkout_new_connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:871:in `acquire_connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:593:in `checkout'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:437:in `connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:1119:in `retrieve_connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_handling.rb:221:in `retrieve_connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_handling.rb:189:in `connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/postgresql_database_tasks.rb:12:in `connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/postgresql_database_tasks.rb:21:in `create'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:126:in `create'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:185:in `block in create_current'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:479:in `block (2 levels) in each_current_configuration'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:476:in `each'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:476:in `block in each_current_configuration'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:475:in `each'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:475:in `each_current_configuration'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:184:in `create_current'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/railties/databases.rake:39:in `block (2 levels) in <main>'
/usr/local/bundle/gems/bugsnag-6.11.1/lib/bugsnag/integrations/rake.rb:18:in `execute_with_bugsnag'

Caused by:
PG::ConnectionBad: FATAL:  role "postgres" does not exist
/usr/local/bundle/gems/pg-1.1.4/lib/pg.rb:56:in `initialize'
/usr/local/bundle/gems/pg-1.1.4/lib/pg.rb:56:in `new'
/usr/local/bundle/gems/pg-1.1.4/lib/pg.rb:56:in `connect'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb:46:in `postgresql_connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:887:in `new_connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:931:in `checkout_new_connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:910:in `try_to_checkout_new_connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:871:in `acquire_connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:593:in `checkout'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:437:in `connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:1119:in `retrieve_connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_handling.rb:221:in `retrieve_connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/connection_handling.rb:189:in `connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/postgresql_database_tasks.rb:12:in `connection'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/postgresql_database_tasks.rb:21:in `create'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:126:in `create'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:185:in `block in create_current'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:479:in `block (2 levels) in each_current_configuration'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:476:in `each'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:476:in `block in each_current_configuration'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:475:in `each'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:475:in `each_current_configuration'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/tasks/database_tasks.rb:184:in `create_current'
/usr/local/bundle/gems/activerecord-6.0.3.2/lib/active_record/railties/databases.rake:39:in `block (2 levels) in <main>'
/usr/local/bundle/gems/bugsnag-6.11.1/lib/bugsnag/integrations/rake.rb:18:in `execute_with_bugsnag'
Tasks: TOP => db:create
(See full trace by running task with --trace)
```
even though user roles and databases already exist in postgresql

```
root=# \l
                                  List of databases
   Name    |  Owner   | Encoding |   Collate   |    Ctype    |   Access privileges
-----------+----------+----------+-------------+-------------+-----------------------
 postgres  | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =Tc/postgres         +
           |          |          |             |             | postgres=CTc/postgres+
           |          |          |             |             | root=c/postgres
 root      | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 |
 template0 | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/postgres          +
           |          |          |             |             | postgres=CTc/postgres
 template1 | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/postgres          +
           |          |          |             |             | postgres=CTc/postgres
(4 rows)
```

Why do I feel this software is the most difficult to install that I have ever used because it never installs successfully 😢

Then I try back to heroku build again, the problem is that now I can't log in after successful install, there isn't any message that appears besides the "error" message
![Screenshot_2](https://user-images.githubusercontent.com/53988458/88764895-488afa80-d1a8-11ea-83ec-e22e0b66fcd7.png)

![Screenshot_3](https://user-images.githubusercontent.com/53988458/88765224-bdf6cb00-d1a8-11ea-80b4-b090b3253d47.png)

**Comment by michelson** [comment]
Hi , in heroku try to remove all the cookies , maybe your old version mess
up something with the new one ? hard to tell

**Comment by kang28ivan** [comment]
> Hi , in heroku try to remove all the cookies , maybe your old version mess up something with the new one ? hard to tell

I have done this, in myoku I also use a different application name, I do not understand why previously installed on heroku successfully using the same .env configuration.
My other question is on Docker and ubuntu, I built on all versions of Ubuntu from 14.04 to 20.04 even Debian nothing works, the error as above, Is there no production version to install on Ubuntu?

**Comment by bensssyde** [comment]
Hello,

for 
FATAL:  role "postgres" does not exist 

try in this section of docker-compose file
``` 
environment:
      POSTGRES_USER: postgresql
```

    
change from postgresql to postgres

**Comment by michelson** [comment]
Hi , in heroku did you run the **migrate** & **seed** & **admin_generate** task ?

**Comment by kang28ivan** [comment]
> Hello,
> 
> for
> FATAL: role "postgres" does not exist
> 
> try in this section of docker-compose file
> 
> ```
> environment:
>       POSTGRES_USER: postgresql
> ```
> 
> change from postgresql to postgres

Already, and success install but rails can't start


> Hi , in heroku did you run the **migrate** & **seed** & **admin_generate** task ?

Sure, I following step by step nothing is missed in the documentation.
When I run bundle exec rails server long time to finish, already more than 1 hour I waiting for this 
![Screenshot_2](https://user-images.githubusercontent.com/53988458/88897424-ce2caa00-d27d-11ea-82a1-55aec3bc6693.png)

Also on heroku, several times I reinstall never managed to login and error

> 1 error prohibited this agent from being saved:
> Email not found

Can you explain install for production with docker or ubuntu?

**Comment by svoop** [comment]
Duplicate of #165

**Comment by michelson** [comment]
Hi , I've added some articles on production deployment, one for Caprover and for digital Ocean apps, 
https://dev.chaskiq.io/en/collections/production-configuration

**Comment by svoop** [comment]
Cool, thanks a bunch @michelson!

**Comment by Turab** [comment]
@svoop Thank you for the new docs. But I can not run Chaskiq using the doc for Caprover. It reboots forever starting with an error "Database 'chaskiq' already exists". I tried dropping the database if the initialization script really desperately desired to create the DB itself; which it did. But it still reboots and still complains about database being already existed.

I also tried to set `DISABLE_DATABASE_ENVIRONMENT_CHECK=1` but still no use. It keeps restarting and the UI never loads. Here is a portion of the app log:

```
2021-02-27T11:26:21.972647755Z Database 'chaskiq' already exists
2021-02-27T11:26:34.207423014Z W, [2021-02-27T11:26:34.207006 #9] WARN -- [Bugsnag]: No valid API key has been set, notifications will not be sent
2021-02-27T11:26:42.666136564Z W, [2021-02-27T11:26:42.665658 #12] WARN -- [Bugsnag]: No valid API key has been set, notifications will not be sent
2021-02-27T11:26:48.248560261Z => Booting Puma
2021-02-27T11:26:48.248590884Z => Rails 6.1.2 application starting in production 
2021-02-27T11:26:48.248595606Z => Run `bin/rails server --help` for more startup options
2021-02-27T11:26:52.779394457Z /usr/src/app/app/models/banner.rb:40: warning: key :show_sender is duplicated and overwritten on line 46
......................
2021-02-27T11:26:56.072809195Z from bin/rails:6:in `<main>'
2021-02-27T11:29:22.961538411Z Database 'chaskiq' already exists
```

**Comment by dantosXD** [comment]
I also am running into the issues Turab describes on the caprover One-Click-Install as well as following the production deployment instructions for caprover.

**Comment by jhoedram** [comment]
_Why do I feel this software is the most difficult to install that I have ever used because it never installs successfully 😢_

It will be because Chaskiq has an installation service. And if they offer you the very clear and easy guide, they would be left without the installation earnings.

**Comment by michelson** [comment]
@jhoedram, Chaskiq, it's just a standard Rails app, and anyone could install it, not just me. So what you are implying is just not fair. Also, I've documented three ways to install the platform. Heroku, Caprover and Docker. At the moment, so many people have figured out how to install the platform, there are no tricks there, and I also help those who don't; this issue thread is an example of how I'm not charging anyone a priori and willing to help. 
Said that I also offer my time to speed integration for people that don't want to waste time installing this. I believe that that has to have value. Right?

**Comment by zackrspv** [comment]
> @jhoedram, Chaskiq, it's just a standard Rails app, and anyone could install it, not just me. So what you are implying is just not fair. Also, I've documented three ways to install the platform. Heroku, Caprover and Docker. At the moment, so many people have figured out how to install the platform, there are no tricks there, and I also help those who don't; this issue thread is an example of how I'm not charging anyone a priori and willing to help. Said that I also offer my time to speed integration for people that don't want to waste time installing this. I believe that that has to have value. Right?

If that's truly the case, I would recommend a proper documentation for: 
- Installation for Production With Docker
- Installation for Production with Ubuntu

The other ones you have are fine (but do not work - especially heroki, fails out of the box). 

The documentation just isn't complete; you should have step by step procedures in place so that people, and IT admins, and server admins can see the steps to take (to ensure they are doing things properly). What you have now is scattered throughout your docs, missing critical steps (often requires us to go to 3rd party sites, stack overflow, official rails documentations) to solve issues, and get them working with your instructions. 

SIMPLY PUT: 
YOU. NEED. BETTER. DOCUMENTATION.

**Comment by michelson** [comment]
Hi @zackrspv, so is there a specific error blocking you from deploying?
I think there is already a guide for Heroku, with even a Video on how to do it, step by step: 
https://dev.chaskiq.io/production-configuration/deploy-on-heroku 

Also, the guide for ubuntu is for development, not for production, as the title states: 
https://dev.chaskiq.io/en/articles/install-chaskiq-on-ubuntu-for-development , also are three other guides on how to install this on windows, osx https://dev.chaskiq.io/getting-started

if you have a specific error that is blocking you, you could add a proper issue, so then I could help or even fix the documentation; otherwise is difficult to help

**Comment by zackrspv** [comment]
> Hi @zackrspv, so is there a specific error blocking you from deploying? I think there is already a guide for Heroku, with even a Video on how to do it, step by step: https://dev.chaskiq.io/production-configuration/deploy-on-heroku
> 
> Also, the guide for ubuntu is for development, not for production, as the title states: https://dev.chaskiq.io/en/articles/install-chaskiq-on-ubuntu-for-development , also are three other guides on how to install this on windows, osx https://dev.chaskiq.io/getting-started
> 
> if you have a specific error that is blocking you, you could add a proper issue, so then I could help or even fix the documentation; otherwise is difficult to help

My Friend, doing the exact steps in your Heroku setup also fails - following your video, also fails. The Heroku app doesn't set it up properly or fails to create at all. That's the point I'm trying to make to you. 

And your ubuntu one is only for DEVELOPMENT, not production (you should provide one for production for those who don't like to use Docker).  And, you should provide one for DOCKER as well, so that people know what to expect, and how to modify, start/stop, etc this behemoth of an amazing product. 

It wouldn't be hard for you to take a day, sit down, and write step by step instructions - the vast majority of open source GitHub projects do such things - because they know a) not everyone does the same thing you do, b) not everyone uses the same versions of things you do and c) not everyone has the same level of experience you do. 

Seeing the steps written out for production affords HUGE benefits not only for YOU, to ensure people are doing things correctly, removes additional roadblocks for server admins from having to write issues on your github pages, and allows IT/Server/Project members to ensure everyone is aligned with what needs to be ran, and why. 

Even AWS, Microsoft and GitHub provide step by step instructions - heck even Chatwoot does. Y'all just need to sit down and write those.

**Comment by michelson** [comment]
Hi, 

When you say:
> setup also fails - following your video, also fails. 

It does not add any value to the conversation, please provide a trace or evidence on which step are you stuck, so we can provide an amendment to the documentation and even help you. There are many people that were able to follow the documentation and install the platform locally or in cloud services.

**Comment by eelco2k** [comment]
Perhaps this might help [#165](https://github.com/chaskiq/chaskiq/issues/165)

and i created a small repo for a docker-compose in production so you can run it on own server:
[https://github.com/eelco2k/chaskiq-docker-compose](https://github.com/eelco2k/chaskiq-docker-compose)

**Comment by michelson** [comment]
thanks @eelco2k !
